src/game/actions.ts

```diff
@@ -12,7 +12,7 @@ export interface MoveAction {
 
 export interface MagicAction {
```

Comment:
Maybe we need to add `requiredMagicLevel` to `Spell` here

Reply:
The `magicLevel` is already in the base class.

One of TypeScript’s core principles is that type checking focuses on the shape that values have. This is sometimes called “duck typing” or “structural subtyping”. In TypeScript, interfaces fill the role of naming these types, and are a powerful way of defining contracts within your code as well as contracts with code outside of your project.